ci-operator/step-registry/ipi/conf/external-lb/ipi-conf-external-lb-commands.sh

```diff
@@ -0,0 +1,266 @@
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+BASTION_IP=$(<"${SHARED_DIR}/bastion_private_address")
+BASTION_SSH_USER=$(<"${SHARED_DIR}/bastion_ssh_user")
+
+haproxy_cfg_filename="haproxy.cfg"
+haproxy_cfg="${SHARED_DIR}/${haproxy_cfg_filename}"
+bastion_haproxy_cfg="/tmp/${haproxy_cfg_filename}"
+master_ips=(`oc get nodes --selector=node-role.kubernetes.io/master -o jsonpath={.items[*].status.addresses[?\(@.type==\"InternalIP\"\)].address}`)
+worker_ips=(`oc get nodes --selector=node-role.kubernetes.io/worker -o jsonpath={.items[*].status.addresses[?\(@.type==\"InternalIP\"\)].address}`)
+
+## HAProxy config
+cat > ${haproxy_cfg} << EOF
+defaults
+  maxconn 20000
+  mode    tcp
+  log     /var/run/haproxy/haproxy-log.sock local0
+  option  dontlognull
+  retries 3
+  timeout http-request 10s
+  timeout queue        1m
+  timeout connect      10s
+  timeout client       86400s
+  timeout server       86400s
+  timeout tunnel       86400s
+
+frontend api-server
+    bind ${BASTION_IP}:6443
+    default_backend api-server
+
+frontend machine-config-server
+    bind ${BASTION_IP}:22623
+    default_backend machine-config-server
+
+frontend router-http
+    bind ${BASTION_IP}:80
+    default_backend router-http
+
+frontend router-https
+    bind ${BASTION_IP}:443
+    default_backend router-https
+
+backend api-server
+    option  httpchk GET /readyz HTTP/1.0
+    option  log-health-checks
+    balance roundrobin
+    $( for ip in "${master_ips[@]}"
+    do
+      echo "    server $ip $ip:6443 weight 1 verify none check check-ssl inter 1s fall 2 rise 3"
+    done
+    )
+
+backend machine-config-server
+    balance roundrobin
+    $( for ip in "${master_ips[@]}"
+    do
+      echo "    server $ip $ip:22623 check"
+    done
+    )
+
+backend router-http
+    balance source
+    mode tcp
+    $( for ip in "${worker_ips[@]}"
+    do
+      echo "    server $ip $ip:80 check"
+    done
+    )
+
+backend router-https
+    balance source
+    mode tcp
+    $( for ip in "${worker_ips[@]}"
+    do
+      echo "    server $ip $ip:443 check"
+    done
+    )
+EOF
+
+# scp new_pull_secret credential to bastion host
+scp -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no "${haproxy_cfg}" ${BASTION_SSH_USER}@${BASTION_IP}:${bastion_haproxy_cfg}
+
+# Create /etc/haproxy/haproxy.cfg and load it by restart haproxy.service in bastion host
+ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} "sudo systemctl restart haproxy.service"
```

Comment:
seems missing step to update /etc/haproxy/haproxy.cfg after copying it to /tmp folder on bastion host.

Reply:
@jinyunma restart haproxy.service will reload /tmp/haproxy.cfg, I need to correct the comment, thanks for the catch.